### PR TITLE
AF-593: Decouple kie-wb-common-datamodel from Drools

### PR DESCRIPTION
**Pull request by mbarkley**

### CODE REVIEW

**Comment by mbarkley** [comment]
Related:
https://github.com/AppFormer/uberfire/pull/822
https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/541
https://github.com/kiegroup/droolsjbpm-knowledge/pull/263
https://github.com/kiegroup/drools/pull/1444
https://github.com/kiegroup/jbpm/pull/973
https://github.com/kiegroup/droolsjbpm-integration/pull/1163
https://github.com/kiegroup/kie-wb-common/pull/1133
https://github.com/kiegroup/drools-wb/pull/596
https://github.com/kiegroup/jbpm-designer/pull/654
https://github.com/kiegroup/jbpm-wb/pull/864

**Review by manstis — APPROVED** [review]

**Comment by manstis** [comment]
@mcimbora Looks OK to me, but appreciate your approval as this is _your_ code!

File: optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-backend/src/test/java/org/optaplanner/workbench/screens/solver/backend/server/TestUtil.java [hunk 1]
```diff
@@ -20,8 +20,16 @@
...
 import java.io.Reader;
 
+import javax.enterprise.inject.Produces;
+
+import org.appformer.project.datamodel.commons.util.MVELEvaluator;
+import org.appformer.project.datamodel.commons.util.RawMVELEvaluator;
+
 public class TestUtil {
 
+    @Produces
```
Comment (mcimbora) [inline optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-backend/src/test/java/org/optaplanner/workbench/screens/solver/backend/server/TestUtil.java]: @mbarkley Is the evaluator required? Just can't see it being injected anywhere around, could you elaborate? Thanks.

Comment (mbarkley) [reply optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-backend/src/test/java/org/optaplanner/workbench/screens/solver/backend/server/TestUtil.java]: @mcimbora Yes the evaluator is injected in several places in kie-wb-common (whereas it used to be created via static methods). Without this Weld gets an exception causing tests in this module to fail.

In the KIE webapps the producer is supplied by `drools-cdi`, so now any tests using Weld and the DMO need to either depend on `drools-cdi` or provide their own evaluator.

**Comment by mbarkley** [comment]
Jenkins, please retest this.

**Comment by mbarkley** [comment]
Jenkins, please retest this.

**Comment by mbarkley** [comment]
Insufficient memory. Trying again.

Jenkins, please retest this.

**Comment by manstis** [comment]
Doing what again? I forget.

**Comment by mbarkley** [comment]
@manstis It ran out of memory GWT-compiling optaplanner-wb-webapp.

**Comment by manstis** [comment]
@mbarkley I know I was attempting to be _funny_.. I should stop trying to be something I am not!

**Comment by mbarkley** [comment]
@manstis Reading that just now I did laugh out loud. Sometimes humour is lost on those knee-deep in PRs :-)

**Review by mcimbora — APPROVED** [review]
@mbarkley Sounds good, thanks.

**Comment by mbarkley** [comment]
Jenkins, please retest this.

**Comment by mbarkley** [comment]
Jenkins, please retest this.

**Comment by mbarkley** [comment]
Jenkins, please retest this.

**Comment by mbarkley** [comment]
Jenkins, please retest this.

**Comment by mbarkley** [comment]
Jenkins, please retest this.

**Pull request merged**